### PR TITLE
Add integration test suite

### PR DESCRIPTION
**Pull request by jeremyjams**
- Logs:
```
14:58:11    Integration tests
14:58:11  Starting services..
[...]
15:04:44      ✔ should get protocol (253ms)
[...]
15:04:44    1 passing (7m)
```
- Main & develop are now protected and both unit & integration test checks are mandatory:
image

### CODE REVIEW

File: docker/test/docker-compose.yml [hunk 1]
```diff
@@ -1,69 +1,72 @@
...
   chain:
-    image: "iexechub/poco-chaintest:5.3.0-token-parity"
+    image: docker-regis.iex.ec/poco-chain:1.0.0-poco-v5.5.0-voucher-v1.0.0-nethermind
     restart: unless-stopped
-    networks:
-      - blockchain
     expose:
       - 8545
-      - 8546
+#      - 8546
```
Comment (jeremyjams) [inline docker/test/docker-compose.yml:7]:
```suggestion
      # - 8546
```

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM node:22
+WORKDIR /iexec-poco-subgraph
+COPY package*.json .
+COPY schema.graphql .
+COPY subgraph.template.yaml .
+COPY networks.json .
+COPY src src
+RUN npm ci
```
Comment (zguesmi) [inline docker/Dockerfile:8]: Put `npm ci` right after line 3 to use Docker cache.

File: docker/test/docker-compose.yml [hunk 3]
```diff
@@ -1,69 +1,72 @@
...
 
   ipfs:
-    image: ipfs/go-ipfs:v0.10.0
     restart: unless-stopped
-    networks:
-      - thegraph
+    image: ipfs/go-ipfs:v0.22.0
+    expose:
+      - 8080
+      - 5001
```
Comment (zguesmi) [inline docker/test/docker-compose.yml:17]:
```suggestion
```
No need for expose if you use `ports` here.

Comment (jeremyjams) [reply docker/test/docker-compose.yml]: Sure see [Removed exposed ports which already mapped externally](https://github.com/iExecBlockchainComputing/PoCo-subgraph/pull/21/commits/47f83b650cfadfe9cb364f43e704de8ba377d1cc) :)

File: docker/Dockerfile (hunk 2 above)
Comment (jeremyjams) [reply docker/Dockerfile]: Yes thanks there it is [Move npm install to lower layer](https://github.com/iExecBlockchainComputing/PoCo-subgraph/pull/21/commits/d8ef02f6b4f4ec4cbba8e64b7bba69acd3cdc46f)!

**Review by zguesmi — APPROVED** [review]
Well done!

**Review by gfournieriExec — APPROVED** [review]

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -20,3 +20,4 @@ jobs:
         run: npm run test
       - name: Test build
         run: npm run build
+    # See Jenkinsfile-itest for "Run integration tests" step
```
Comment (jeremyjams) [inline .github/workflows/main.yml:23]: @DamienMure FYI 1

File: Jenkinsfile-itest [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+            try {
+                sh "echo -n '${password}' | docker login --username '${username}' --password-stdin ${registry}"
+                sh 'cd docker/test/ && docker compose pull chain'
+            } finally {
+                sh "docker logout ${registry}"
+            }
+        }
+    }
+    docker.image('node:22-alpine')
+        .inside('-v /var/run/docker.sock:/var/run/docker.sock --network=host --user=root') {
```
Comment (jeremyjams) [inline Jenkinsfile-itest:22]: @DamienMure FYI 2

**Pull request merged**